arch/arc/core/fault.c

```diff
@@ -123,6 +123,208 @@ static u32_t z_check_thread_stack_fail(const u32_t fault_addr, u32_t sp)
+		printk("EV_DCError");
+		break;
+	case ARC_EV_MISALIGNED:
+		printk("EV_Misaligned");
+		break;
+	case ARC_EV_VEC_UNIT:
+		printk("EV_VecUnit");
+		break;
+	default:
+		printk("unknown vector");
+		break;
+	}
+}
+
+/* For EV_ProtV, the numbering/semantics of the parameter are consistent across
+ * several codes, although not all combination will be reported.
+ *
+ * These codes and parameters do not have associated* names in
+ * the technical manual, just switch on the values in Table 6-5
+ */
+static void dump_protv_access_err(u32_t parameter)
+{
+	switch (parameter) {
+	case 0x1:
+		printk("code protection scheme");
+		break;
+	case 0x2:
+		printk("stack checking scheme");
+		break;
+	case 0x4:
+		printk("MPU");
+		break;
+	case 0x8:
+		printk("MMU");
+		break;
+	case 0x10:
+		printk("NVM");
+		break;
+	case 0x24:
+		printk("Secure MPU");
+		break;
+	case 0x44:
+		printk("Secure MPU with SID mismatch");
+		break;
+	default:
+		printk("unknown");
+		break;
+	}
+}
+
+static void dump_protv_exception(u32_t code, u32_t parameter)
+{
+	switch (code) {
+	case 0x0:
+		printk("Instruction fetch violation: ");
+		dump_protv_access_err(parameter);
+		break;
+	case 0x1:
+		printk("Memory read protection violation: ");
+		dump_protv_access_err(parameter);
+		break;
+	case 0x2:
+		printk("Memory write protection violation: ");
+		dump_protv_access_err(parameter);
+		break;
+	case 0x3:
+		printk("Memory read-modify-write violation: ");
+		dump_protv_access_err(parameter);
+		break;
+	case 0x10:
+		printk("Normal vector table in secure memory");
+		break;
+	case 0x11:
+		printk("NS handler code located in S memory");
+		break;
+	case 0x12:
+		printk("NSC Table Range Violation");
+		break;
+	default:
+		printk("unknown exception code");
+		break;
+	}
+}
+
+static void dump_machine_check_exception(u32_t code, u32_t parameter)
+{
+	switch (code) {
+	case 0x0:
+		printk("double fault");
+		break;
+	case 0x1:
+		printk("overlapping TLB entries");
+		break;
+	case 0x2:
+		printk("fatal TLB error");
+		break;
+	case 0x3:
+		printk("fatal cache error");
+		break;
+	case 0x4:
+		printk("internal memory error on instruction fetch");
+		break;
+	case 0x5:
+		printk("internal memory error on data fetch");
+		break;
+	case 0x6:
+		printk("illegal overlapping MPU entries");
+		if (parameter == 0x1) {
+			printk(" (jump and branch target)");
+		}
+		break;
+	case 0x10:
+		printk("secure vector table not located in secure memory");
+		break;
+	case 0x11:
+		printk("NSC jump table not located in secure memory");
+		break;
+	case 0x12:
+		printk("secure handler code not located in secure memory");
+		break;
+	case 0x13:
+		printk("NSC target address not located in secure memory");
+		break;
+	case 0x80:
+		printk("uncorrectable ECC or parity error in vector memory");
+		break;
+	default:
+		printk("unknown exception code");
+		break;
+	}
+}
+
+static void dump_exception_info(u32_t vector, u32_t code, u32_t parameter)
+{
+	dump_vector_name(vector);
+
+	switch (vector) {
```

Comment:
This switch could be merged into dump_vector_name(), that has the switch on vector number already.

Reply:
done